x-pack/plugins/data_visualizer/public/application/data_drift/document_count_chart_single_brush/document_count_chart_singular.tsx

```diff
@@ -0,0 +1,499 @@
+      return [
+        ...chartPointsSplit,
+        {
+          time: interval ? Number(chartPointsSplit[0].time) + interval : timeRangeEarliest,
+          value: 0,
+        },
+      ];
+    }
+    return chartPointsSplit;
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [chartPointsSplit, timeRangeEarliest, timeRangeLatest, interval]);
+
+  const snapTimestamps = useMemo(() => {
+    const timestamps: number[] = [];
+    let n = timeRangeEarliest;
+
+    while (n <= timeRangeLatest + interval) {
+      timestamps.push(n);
+      n += interval;
+    }
+
+    return timestamps;
+  }, [timeRangeEarliest, timeRangeLatest, interval]);
+
+  const timefilterUpdateHandler = useCallback(
+    (range: TimeFilterRange) => {
+      data.query.timefilter.timefilter.setTime({
+        from: moment(range.from).toISOString(),
+        to: moment(range.to).toISOString(),
+        mode: 'absolute',
+      });
+    },
+    [data]
+  );
+
+  const onBrushEnd = ({ x }: XYBrushEvent) => {
+    if (!x) {
+      return;
+    }
+    const [from, to] = x;
+    timefilterUpdateHandler({ from, to });
+  };
+
+  const timeZone = getTimeZone(uiSettings);
+
+  const [originalWindowParameters, setOriginalWindowParameters] = useState<
+    SingleBrushWindowParameters | undefined
+  >();
+  const [windowParameters, setWindowParameters] = useState<
+    SingleBrushWindowParameters | undefined
+  >();
+
+  const triggerAnalysis = useCallback(
+    (startRange: number | SingleBrushWindowParameters) => {
+      if (viewMode === VIEW_MODE.ZOOM && typeof startRange === 'number') {
+        const range: TimeFilterRange = {
+          from: startRange,
+          to: startRange + interval,
+        };
+
+        timefilterUpdateHandler(range);
+      } else if (viewMode === VIEW_MODE.BRUSH) {
+        if (
+          originalWindowParameters === undefined &&
+          windowParameters === undefined &&
+          adjustedChartPoints !== undefined
+        ) {
+          const wp =
+            typeof startRange === 'number'
+              ? getSingleBrushWindowParameters(
+                  startRange + interval / 2,
+                  timeRangeEarliest,
+                  timeRangeLatest + interval
+                )
+              : startRange;
+          const wpSnap = getSnappedSingleBrushWindowParameters(wp, snapTimestamps);
+          setOriginalWindowParameters(wpSnap);
+          setWindowParameters(wpSnap);
+
+          if (brushSelectionUpdateHandler !== undefined) {
+            brushSelectionUpdateHandler(wpSnap, true);
+          }
+        }
+      }
+    },
+    [
+      interval,
+      timeRangeEarliest,
+      timeRangeLatest,
+      snapTimestamps,
+      originalWindowParameters,
+      setWindowParameters,
+      brushSelectionUpdateHandler,
+      adjustedChartPoints,
+      timefilterUpdateHandler,
+      viewMode,
+      windowParameters,
+    ]
+  );
+
+  const onElementClick: ElementClickListener = useCallback(
```

Comment:
I'm finding that the range selected on click doesn't coincide with the clicked on histogram bucket. Is this the expected behavior? Currently I always have to adjust the range that is created on click.  

![data_drift_selection](https://github.com/elastic/kibana/assets/7405507/a6db8250-4721-491a-b98c-ba5aac5586ad)